### ISSUE TITLE
panic: invalid character ']' while importing to corpus

### ISSUE THREAD

**Issue by carstenhag** [lidel/meowkov#14]
```
carstenh@archmoter8 ~/g/meowkov> cat log.log | sudo make docker-corpus-add
docker start meowkov_corpus
meowkov_corpus
docker start meowkov_irc
meowkov_irc
docker run -v /home/carstenh/git/meowkov/meowkov.conf:/meowkov/meowkov.conf:ro --link meowkov_corpus:redis -i --rm meowkov -import=true -purge=false
panic: invalid character ']' looking for beginning of value

goroutine 1 [running]:
main.check(0x7f03aec0cd40, 0xc20801e240)
    /meowkov/meowkov.go:636 +0x50
main.loadConfig(0x6b4d30, 0xc, 0x0)
    /meowkov/meowkov.go:83 +0x312
main.main()
    /meowkov/meowkov.go:129 +0x34

goroutine 2 [runnable]:
runtime.forcegchelper()
    /usr/src/go/src/runtime/proc.go:90
runtime.goexit()
    /usr/src/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 3 [runnable]:
runtime.bgsweep()
    /usr/src/go/src/runtime/mgc0.go:82
runtime.goexit()
    /usr/src/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 4 [runnable]:
runtime.runfinq()
    /usr/src/go/src/runtime/malloc.go:712
runtime.goexit()
    /usr/src/go/src/runtime/asm_amd64.s:2232 +0x1
^C^C^C
```

My standard 43MB log.  Not sure why this happens. Trying out ~2 week old meowkov code makes the same bug appear.

Even cat'ing an empty file into it causes this error.

I've upgraded packages recently, that may have caused it?

**Comment by carstenhag** [comment]
http://i.imgur.com/9h1xp91.png - output of `sudo docker daemon`, after trying to ^C^C the corpus add command.

**Comment by lidel** [comment]
The `invalid character ']' looking for beginning of value` looks like an error returned by JSON parser. Check if your `meowkov.conf` is a valid JSON (eg. at http://jsoneditoronline.org/), maybe there is a typo :-)

I just improved error reporting (037d9b3): please run `git pull && make docker-rebuild` (it will remove any hanging containers), try again and let me know if it helped.

**Comment by carstenhag** [comment]
Hmmm. It should error before (or during) docker-rebuild/docker-start though, right? Right now it rebuilds/starts without any error, it just doesn't connect to my server.

**Comment by carstenhag** [comment]
But yes, I had a trailing `,` in the smileys array, so used to Go wanting that :100: 

Thanks for the improvement!

**Comment by lidel** [comment]
Glad it helped.
On ending note: `docker-start` runs container in detached mode (in background), so it probably failed silently.  To see error you would have to display output via `make docker-logs`.

A permanent fix will be implemented by #15.
It should make config-related issues should easier to spot  :-)
